concent_api/signing_service_testing_common.py

```diff
@@ -0,0 +1,236 @@
+    parser.add_argument(
+        '--ethereum-blockchain-geth-address',
+        dest='ethereum_blockchain_geth_address',
+        type=str,
+    )
+    parser.add_argument(
+        '--ethereum-blockchain-concent-ethereum-address',
+        dest='ethereum_blockchain_concent_ethereum_address',
+        type=str,
+    )
+
+    return parser.parse_args()
+
+
+def update_config(config, arguments):
+    """ Updates values in config on arguments parsed from command line. """
+    if arguments.concent_api_url is not None:
+        config[Components.CONCENT_API.value]['api_url'] = arguments.concent_api_url
+
+    if arguments.middleman_host is not None:
+        config[Components.MIDDLEMAN.value]['host'] = arguments.middleman_host
+    if arguments.middleman_port is not None:
+        config[Components.MIDDLEMAN.value]['port'] = arguments.middleman_port
+
+    if arguments.ethereum_blockchain_geth_address is not None:
+        config[Components.ETHEREUM_BLOCKCHAIN.value]['geth_address'] = arguments.ethereum_blockchain_geth_address
+    if arguments.ethereum_blockchain_concent_ethereum_address is not None:
+        config[Components.ETHEREUM_BLOCKCHAIN.value]['concent_ethereum_address'] = arguments.ethereum_blockchain_concent_ethereum_address
+
+    return config
+
+
+def are_required_settings_in_config(components, config):
+    """ Verifies if config values for required components are set. """
+    for component in components:
+        assert config.has_section(component.value)
+        assert config.options(component.value) == REQUIRED_COMPONENTS_SETTINGS[component]
+
+
+def run_tests(objects):
+    """ Main tests runner, prepares test config, gather test functions, executes them, show results. """
+    assert 'REQUIRED_COMPONENTS' in objects, 'Test suite must define `REQUIRED_COMPONENTS`.'
+
+    # Parse config.
+    config = read_config()
+    arguments = read_command_line()
+    config = update_config(config, arguments)
+    are_required_settings_in_config(objects['REQUIRED_COMPONENTS'], config)
+
+    # Check if required components are running.
+    check_if_required_components_are_running(objects['REQUIRED_COMPONENTS'], config)
+
+    # Prepare tests list.
+    test_id = str(random.randrange(1, 100000))
+    tests_to_execute = get_tests_list([], list(objects.keys()))
+    print("Tests to be executed: \n * " + "\n * ".join(tests_to_execute))
+    print()
+
+    # Execute tests.
+    execute_tests(
+        tests_to_execute=tests_to_execute,
+        objects=objects,
+        test_id=test_id,
+        config=config,
+    )
+
+    # Count fails.
+    if count_fails.get_fails() > 0:
+        count_fails.print_fails()
+
+    print("END")
+
+
+def check_if_required_components_are_running(components, config):
+    """ Main tests runner, prepares test config, gather test functions, executes them, show results. """
+    if Components.CONCENT_API in components:
+        if not is_concent_api_is_running(**dict(config.items(Components.CONCENT_API.value))):
```

Comment:
`is_concent_api_is_running` -> `is_concent_api_running`

Reply:
Those were already done, you must have not refreshed the page today ;)